### ISSUE TITLE
include more info in beacon hashes

### ISSUE THREAD

**Issue by joelkoen** [beacondb/beacondb#1]
[Surveilling the Masses with Wi-Fi-Based Positioning Systems](https://arxiv.org/abs/2405.14975) mentions how incredibly easy it is to get a list of MAC addresses in an area from Apple's WPS DB. Currently beacon hashes include SSIDs, but after collecting a lot more data I'm finding that a significant amount of default access point names are derived from the BSSID.

I would like to include more information in the beacon hash to make it even more difficult for someone to abuse the nature of such a large dataset. It might be worth MITMing some phone WPS submissions to choose some kind of data point that isn't currently being collected, and add that to the beacon hash.

**Comment by joelkoen** [comment]
The paper also discusses spoofing APs from another country to make a device appear in that location, another reason why this information shouldn't be so easily accessible.

**Comment by joelkoen** [comment]
I've built a basic POC on fingerprinting APs based on the IE (Information Elements) they advertise: https://github.com/beacondb/bcap/blob/main/src/beacon.rs

I've decided not to go ahead with this as I simply don't have enough data to build something I can be confident in, especially because hashes are not reversible. I found that the WPS IE includes a UUID E identifier, which could be another useful "uniqueness" source, but I found online that this may have been derived from the device's physical MAC anyway.

I feel like the SSID is always going to be the best "proof I physically saw it" as it is customizable by humans, meaning it would be difficult to guess combined with the MAC.

I also found that APs broadcast their uptime in the beacon's timestamp field, which could be used to prevent abuse in the future, but Android does not seem to expose an API to access this.
